cfgov/unprocessed/js/modules/VideoPlayer.js

```diff
@@ -0,0 +1,206 @@
+         iFrameElement === null ) {
+      iFrameElement = _createElement( 'iframe', _this.iFrameProperties );
+      iFrameElement.classList.add(  CLASSES.IFRAME_CLASS_NAME );
+      iFrameElement.setAttribute('frameborder', 0 );
+      state.isIframeLoading = true;
+      iFrameElement.onload = function iframeLoad() { state.isIframeLoaded = true; };
+      iFrameContainerElement.appendChild( iFrameElement );
+    } else if( _this.childElements.iframeContainer === null ) {
+      throw new Error( 'No iframe container element found.' );
+    }
+
+    return iFrameElement;
+}
+
+function _ensureElement( domElement, createElementFlag, msg ) {
+  if( domElement instanceof HTMLElement === false ) {
+    msg = msg || DOM_INVALID;
+    throw new Error( msg );
+  }
+
+  return domElement;
+}
+
+function _initEvents() {
+  _this.childElements.playBtn.addEventListener( 'click', _this.play.bind( _this ) );
+  _this.childElements.closeBtn.addEventListener( 'click', _this.stop.bind( _this ) );
+}
+
+function _setChildElements( elements ) {
+  var elementName;
+  var hasOwnProperty = Object.hasOwnProperty.bind( _this.childElements );
+  var querySelector = _this.baseElement.querySelector.bind( _this.baseElement );
+
+  for ( elementName in elements ) {
+    if ( hasOwnProperty( elementName ) ) {
+      elements[elementName] = querySelector( elements[elementName] );
+    }
+  }
+
+  return elements;
+}
+
+// Public functions and properties.
+var API = {
+
+  baseElement: null,
+
+  childElements: {
+    iFrame:          CLASSES.IFRAME_SELECTOR,
+    iFrameContainer: CLASSES.IFRAME_CONTAINER_SELECTOR,
+    playBtn:         CLASSES.PLAY_BTN_SELECTOR,
+    closeBtn:        CLASSES.CLOSE_BTN_SELECTOR
+  },
+
+  destroy: function destroy() {
+    childElements.playBtn.removeEventListener( 'click', play );
+    childElements.closeBtn.removeEventListener( 'click', stop );
+  },
+
+  embedScript: function( script, callback ) {
+    jsLoader.loadScript( script || this.SCRIPT_API, callback || _noopFunct );
+  },
+
+  init: _noopFunct,
+
+  iFrameProperties : {
+    allowfullscreen:       'true',
+    scrolling:             'no',
+    src:                   '',
+    width:                 '100%'
+  },
+
+  state : {
+    isIframeLoading: false,
+
+    get isIframeLoaded() {
+      return _isIframeLoaded || false;
+    },
+    set isIframeLoaded( value ) {
+      if ( value === true ) _isIframeLoading = false;
+      _isIframeLoaded = value;
+    },
+    get isVideoPlaying() {
+      return _isVideoPlaying || false;
+    },
+    set isVideoPlaying( value ) {
+      if ( value === true ) _isVideoStopped = false;
+      _isVideoPlaying = value;
+    },
+    get isVideoStopped() {
+      return _isVideoStopped || true;
+    },
+    set isVideoStopped( value ) {
+      if ( value === true ) _isVideoPlaying = false;
+      _isVideoStopped = value;
+    }
+  },
+
+  play: function play ( event ) {
+    if ( event ) {
```

Comment:
fixed in https://github.com/cfpb/cfgov-refresh/pull/1625/commits/36f6195becc4c28009ba7b84c31290ffe07277ae